modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/SequenceSpec.scala

```diff
@@ -61,6 +62,14 @@ class SequenceSpec extends FlatSpec {
     )
   )
 
+  def runToCompletion(q: scalaz.stream.async.mutable.Queue[Event], s0: EngineState): EngineState = {
+    def isFinished(status: SequenceState): Boolean =
+      status == Idle || status == SequenceState.Completed || status === Error
+
+    q.enqueueOne(start(seqId)).flatMap( _ =>
+       processE(q).drop(1).takeThrough(a => !isFinished(a._2.get(seqId).get.status) ).runLast.eval(s0)).unsafePerformSync.get._2
```

Comment:
I also tried with `takeWhile` some time ago but gave up after much headache. How did you find `takeThrough`? By looking at scalaz-streams source code? Is the rendered API documentation for `scalaz-stream` public somewhere?